pkg/performanceprofile/profilecreator/profilecreator.go

```diff
@@ -677,76 +677,92 @@ func contains(s []string, str string) bool {
 	return false
 }
 
-// EnsureNodesHaveTheSameHardware returns an error if all the input nodes do not have the same hardware configuration
-func EnsureNodesHaveTheSameHardware(nodeHandlers []*GHWHandler) error {
+// EnsureNodesHaveTheSameHardware returns an error if all the input nodes do not have the same hardware configuration;
+// if core IDs numbering was found different on same NUMA cells of different compute nodes, the function will return true
+// so this be printed out as a warning to the user later
+func EnsureNodesHaveTheSameHardware(nodeHandlers []*GHWHandler) (bool, error) {
```

Comment:
I'm more and more convinced that every time we use a bool, there's a better approach hidden somewhere. Can we think of a new type, maybe a struct, maybe just a set of strings, which embeds the wrong-but-tolerable (warnings?) we found about a topology, and use it everywhere instead of bools?

Reply:
thanks for raising. I agree, that's a better idea I'd be happy to update